### PR TITLE
[otlp exporter] Validate exporter endpoint has port

### PR DESCRIPTION
**Pull request by atmask**
**Description:** This PR updates the otlp exporter config validation to ensure that the "endpoint" specified for the exporter includes a port. The goal of this is to fail fast if the configuration is invalid instead of waiting for an error to arise. The PR adds a function to the ClientConfig defined in configgrpc that parses the port defined in the endpoint. The otlp exporter uses this port parsing to validate that 

**Link to tracking Issue:** [Issue 9505](https://github.com/open-telemetry/opentelemetry-collector/issues/9505)

**Testing:** 

configgrpc:
- Add unit tests to validate ports are correctly parsed in configrpc and that an error is returned if the port is omitted

otlpexporter:
- Add unit test for conifg validation function to ensure that the validation fails for missing endpoint or missing port and that a well-formed endpoint does not fail

### CODE REVIEW

File: exporter/otlpexporter/config.go
Comment (TylerHelmuth) [inline exporter/otlpexporter/config.go:30]: The logic inside these functions is simple enough that we can keep everything inside `Validate`.

File: config/configgrpc/configgrpc.go
Comment (TylerHelmuth) [inline config/configgrpc/configgrpc.go:175]: We're working towards GA for `configgrpc` right now and I am hesitant to add new public APIs. For now, let's move this function into the otlpexporter.

File: exporter/otlpexporter/config.go
Comment (atmask) [reply exporter/otlpexporter/config.go]: Addressed in ec2b9d32

File: config/configgrpc/configgrpc.go
Comment (atmask) [reply config/configgrpc/configgrpc.go]: Addressed in ec2b9d3

File: exporter/otlpexporter/config.go
Comment (bogdandrutu) [inline exporter/otlpexporter/config.go:32]: We have a default port, why not using that if not provided?

File: exporter/otlpexporter/config.go
Comment (TylerHelmuth) [inline exporter/otlpexporter/config.go:41]:
```suggestion
	_, port, err := net.SplitHostPort(c.SanitizedEndpoint())
```

File: exporter/otlpexporter/config.go
Comment (TylerHelmuth) [reply exporter/otlpexporter/config.go:39]: @atmask ya that is normally the process - go update contrib and we'll get that merged to unblock this PR.

File: exporter/otlpexporter/config_test.go
Comment (TylerHelmuth) [inline exporter/otlpexporter/config_test.go:118]: Instead of a new function we normally test the config via `TestUnmarshalConfig`/`TestUnmarshalInvalidConfig`.  You can add new test scenarios to `testdata/` and then add scenarios to the appropriate test function.

Comment (atmask) [reply exporter/otlpexporter/config_test.go]: I see, that make sense. Moved the new test scenario into the testdata

**Comment by atmask** [comment]
Related PR from otel-collector-contrib to update loadbalancingexporter for the change in otlp config validation: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31425/files

File: exporter/otlpexporter/config.go
Comment (atmask) [reply exporter/otlpexporter/config.go]: PR for otel-collector-contrib linked below

File: exporter/otlpexporter/testdata/invalid_configs.yaml
Comment (iblancasa) [inline exporter/otlpexporter/testdata/invalid_configs.yaml:57]: Would it make sense to test an endpoint using a wrong port? Like `example.com:stuff` instead of `example.com:8080`

Comment (atmask) [reply exporter/otlpexporter/testdata/invalid_configs.yaml:73]: Addressed in latest push. Resolving

**Review by iblancasa — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Comment by atmask** [comment]
Just updated to fix the lint issue for new commits going into main and rebased on main. @TylerHelmuth could you please review this or poke a codeowner to push this PR through so that it can get closed off?

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
I think using the default port is better, but we can add the validation for now since it's not working without specifying the port explicitly anyway. 

I think we don't have clear guidance on how components should mutate config structs passed to them when it's required, like in this case. It'd be great to have an issue for that so we can track it separately.

**Pull request merged**
